### DISCUSSION TITLE
nc.Request `nats: timeout`

### DISCUSSION THREAD

**Discussion by tamis-laan** [nats-io/nats.go#1628] [Q&A]
I'm trying to expose an ML model using nats python and calling it through the go client:
``` go
encoded, err = nc.Request("nlp.sec", sentences.Data, time.Second*4)
```
However I get the error `nats: timeout`. 

Using the nats comandline I can make a successful request:
``` bash
nats request nlp.sec '["I am the first sentence.", "I am the second"]'
```
and it returns an answer under 1 second. So I'm not sure where this timeout is comming from. The timeout is to 4 seconds on the request and I have set the timeout on the client to 10 seconds:

``` go
nclient, err = nats.Connect(nats.DefaultUrl,
	nats.Timeout(10*time.Second),
	nats.ReconnectWait(5*time.Second),
)
```

So I'm not sure what this timeout is refering to...

**Comment by derekcollison** [comment]
Just some things to double check.

Make sure your program is bound to the same account and the NATS cli. Also check that the payload you send from your program is not crashing the service.

**Reply by tamis-laan** [reply]
> Make sure your program is bound to the same account and the NATS cli

Yes, I'm running on kubernetes with the official chart, all other clients subscribing to NATS work correctly.

> Also check that the payload you send from your program is not crashing the service.

I checked the logs and there are no error messages. NATS is also not crashing and restarted by kubernetes.

Does this mean it's a bug?

**Reply by derekcollison** [reply]
Not likely at the core level.

Under NATS cli do `nats account info` and when you do the request add in `--trace`
